// @postago/shared/src/schemas/parcel.ts
import { z } from "zod";

export const createParcelSchema = z.object({
  routeId: z.string().uuid(),
  description: z.string().max(500).optional(),
  declaredValue: z.number().nonnegative().optional(),
  weightKg: z.number().positive().max(1000).optional(),
  lengthCm: z.number().int().positive().max(300).optional(),
  widthCm: z.number().int().positive().max(300).optional(),
  heightCm: z.number().int().positive().max(300).optional(),
});

export const updateParcelWeightSchema = z.object({
  weightKg: z.number().positive().max(1000),
  lengthCm: z.number().int().positive().max(300).optional(),
  widthCm: z.number().int().positive().max(300).optional(),
  heightCm: z.number().int().positive().max(300).optional(),
});

export const updateParcelStatusSchema = z.object({
  status: z.enum([
    "created",
    "paid",
    "received_at_origin",
    "weighed",
    "in_batch",
    "shipped",
    "in_transit",
    "customs",
    "received_at_destination",
    "sorting",
    "out_for_delivery",
    "delivered",
  ]),
  location: z.string().max(200).optional(),
  note: z.string().max(500).optional(),
});

export type CreateParcelInput = z.infer<typeof createParcelSchema>;
export type UpdateParcelWeightInput = z.infer<typeof updateParcelWeightSchema>;
export type UpdateParcelStatusInput = z.infer<typeof updateParcelStatusSchema>;
