// @postago/shared/src/enums.ts
export const UserRole = {
  CLIENT: "client",
  OPERATOR_ORIGIN: "operator_origin",
  OPERATOR_DESTINATION: "operator_destination",
  COURIER: "courier",
  ADMIN: "admin",
} as const;

export type UserRole = (typeof UserRole)[keyof typeof UserRole];

export const ParcelStatus = {
  CREATED: "created",
  PAID: "paid",
  RECEIVED_AT_ORIGIN: "received_at_origin",
  WEIGHED: "weighed",
  IN_BATCH: "in_batch",
  SHIPPED: "shipped",
  IN_TRANSIT: "in_transit",
  CUSTOMS: "customs",
  RECEIVED_AT_DESTINATION: "received_at_destination",
  SORTING: "sorting",
  OUT_FOR_DELIVERY: "out_for_delivery",
  DELIVERED: "delivered",
} as const;

export type ParcelStatus = (typeof ParcelStatus)[keyof typeof ParcelStatus];

export const TransportType = {
  AIR: "air",
  RAIL: "rail",
  SEA: "sea",
  COMBINED: "combined",
} as const;

export type TransportType = (typeof TransportType)[keyof typeof TransportType];

export const BatchStatus = {
  FORMING: "forming",
  SHIPPED: "shipped",
  IN_TRANSIT: "in_transit",
  RECEIVED: "received",
} as const;

export type BatchStatus = (typeof BatchStatus)[keyof typeof BatchStatus];

export const PaymentProvider = {
  CLICK: "click",
  PAYME: "payme",
  KASPI: "kaspi",
  CASH: "cash",
} as const;

export type PaymentProvider =
  (typeof PaymentProvider)[keyof typeof PaymentProvider];

export const PaymentStatus = {
  PENDING: "pending",
  PROCESSING: "processing",
  COMPLETED: "completed",
  FAILED: "failed",
  REFUNDED: "refunded",
} as const;

export type PaymentStatus =
  (typeof PaymentStatus)[keyof typeof PaymentStatus];

export const WarehouseType = {
  ORIGIN: "origin",
  DESTINATION: "destination",
} as const;

export type WarehouseType =
  (typeof WarehouseType)[keyof typeof WarehouseType];

export const NotificationChannel = {
  TELEGRAM: "telegram",
  SMS: "sms",
} as const;

export type NotificationChannel =
  (typeof NotificationChannel)[keyof typeof NotificationChannel];

export const NotificationStatus = {
  PENDING: "pending",
  SENT: "sent",
  FAILED: "failed",
} as const;

export type NotificationStatus =
  (typeof NotificationStatus)[keyof typeof NotificationStatus];

export const Country = {
  KR: "KR",
  CN: "CN",
  TR: "TR",
  UZ: "UZ",
  KZ: "KZ",
} as const;

export type Country = (typeof Country)[keyof typeof Country];

export const COUNTRY_NAMES: Record<string, string> = {
  KR: "Южная Корея",
  CN: "Китай",
  TR: "Турция",
  UZ: "Узбекистан",
  KZ: "Казахстан",
};
